pkg/kapp/permissions/preflight.go

```diff
@@ -37,7 +52,27 @@ func (p *Preflight) SetEnabled(enabled bool) {
 	p.enabled = enabled
 }
 
-func (p *Preflight) SetConfig(_ preflight.CheckConfig) error {
+func (p *Preflight) SetConfig(cfg preflight.CheckConfig) error {
+	pCfg := &PreflightConfig{}
+	cfgBytes, err := json.Marshal(cfg)
+	if err != nil {
+		return fmt.Errorf("converting CheckConfig to bytes: %w", err)
+	}
+
+	err = json.Unmarshal(cfgBytes, pCfg)
+	if err != nil {
+		return fmt.Errorf("parsing permissions preflight config: %w", err)
+	}
+
+	switch pCfg.PermissionValidatorType {
+	case PermissionValidatorTypeSelfSubjectAccessReview:
+	case PermissionValidatorTypeSelfSubjectRulesReview:
```

Comment:
nit: should we add a comment to indicate that the intention is to set a default value and to do validation?
```suggestion
        // Valid, do nothing
	case PermissionValidatorTypeSelfSubjectAccessReview, PermissionValidatorTypeSelfSubjectRulesReview:
```

Reply:
Updated in the latest push